packages/flame/lib/src/collisions/quadtree/quadtree_collision_detection.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:flame/collisions.dart';
+import 'package:flutter/widgets.dart';
+
+/// Collision detection modification to support a Quad Tree broadphase.
+///
+/// Do not use standard [items] list for components. Instead adds all components
+/// into [QuadTreeBroadphase] class.
+class QuadTreeCollisionDetection extends StandardCollisionDetection {
+  QuadTreeCollisionDetection({
+    required Rect mapDimensions,
+    required ExternalBroadphaseCheck onComponentTypeCheck,
+    required ExternalMinDistanceCheck minimumDistanceCheck,
+    int maxObjects = 25,
+    int maxDepth = 10,
+  }) : super(
+          broadphase: QuadTreeBroadphase<ShapeHitbox>(
+            mainBoxSize: mapDimensions,
+            maxObjects: maxObjects,
+            maxDepth: maxDepth,
+            broadphaseCheck: onComponentTypeCheck,
+            minimumDistanceCheck: minimumDistanceCheck,
+          ),
+        );
+
+  QuadTreeBroadphase get quadBroadphase => broadphase as QuadTreeBroadphase;
```

Comment:
Since I think very few (if anyone...) has implemented their own collision detection system I suggest that we instead do a small breaking change in the `CollisionDetection` class so that you can send in a type parameter for the broadphase, so that we don't have to have a separate field for that here.

So the `CollisionDetection` class would look like this instead of how it looks today:
```dart
CollisionDetection<T extends Hitbox<T>, B extends Broadphase<T>> (
  final B broadphase;
  ...
}
```

Reply:
See #1908, what do you think? We can discuss that on that PR.

Reply:
I ready to merge this, as soon as it will be present in "main"

Reply:
Great, now it is merged!

Reply:
done 👌